### PR TITLE
Add OpInfo tests for torch.addbmm

### PR DESCRIPTION
**Pull request by anjali411**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#55378 Add OpInfo tests for torch.addbmm**

Differential Revision: [D27598681](https://our.internmc.facebook.com/intern/diff/D27598681)

### CODE REVIEW

File: test/test_autograd.py [hunk 1]
```diff
@@ -5307,7 +5307,7 @@ def run_functional_checks(test_case, test_name, name, apply_fn, run_grad_checks,
                 'mean', 'inverse', 'addcmul',
                 'addcdiv', 'linalg.tensorinv', 'matrix_exp',
                 'narrow', 'swapaxes', 'swapdims', 'tensor_split',
-                'baddbmm', 'addbmm', 'addmv'] + complex_list_filter + separate_complex_tests
+                'baddbmm', 'addmv'] + complex_list_filter + separate_complex_tests
```
Comment (imaginary-person) [inline test/test_autograd.py:5310]: Hello @anjali411, I'm adding `OpInfo`s for `add`, `mm` and `addmm` via #55201.
Please advise as to whether I should remove `add` from this list in that PR.
Thanks!

Comment (mruberry) [reply test/test_autograd.py:5310]: Why would you remove add, @imaginary-person ? This is for "add B mm".

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -595,6 +595,33 @@ def sample_inputs_addmm(op_info, device, dtype, requires_grad):
...
+                make_tensor(input_args[1], device, dtype,
+                            low=None, high=None,
+                            requires_grad=requires_grad),
+                make_tensor(input_args[2], device, dtype,
+                            low=None, high=None,
+                            requires_grad=requires_grad))
+        alpha, beta = input_args[3], input_args[4]
+        sample_inputs.append(SampleInput(args[0], args=(args[1], args[2]), kwargs=dict(beta=beta, alpha=alpha)))
+        if dtype.is_complex:
+            sample_inputs.append(SampleInput(args[0], args=(args[1], args[2]),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:620]: Cool

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -2366,6 +2393,19 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
                # addmm does not correctly warn when resizing out= inputs
                SkipInfo('TestCommon', 'test_out')),
            sample_inputs_func=sample_inputs_addmm),
+    OpInfo('addbmm',
+           dtypes=floating_types(),
+           dtypesIfCPU=all_types_and_complex_and(torch.float16, torch.bfloat16),
+           dtypesIfCUDA=floating_types_and(torch.float16, torch.complex64, torch.complex128,
+                                           *[torch.bfloat16] if CUDA11OrLater else []),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:2400]: Nice Python

**Review by mruberry — APPROVED** [review]
Awesome!

File: test/test_autograd.py (hunk 1 above)
Comment (imaginary-person) [reply test/test_autograd.py:5310]: @mruberry, I'm adding an `OpInfo` for `add` in #55201, so I was asking if I should remove `add` from this list in that PR.

Comment (mruberry) [reply test/test_autograd.py:5310]: Haha, sorry, that makes much more sense, @imaginary-person! Good question. @anjali411?

Comment (anjali411) [reply test/test_autograd.py:5310]: @imaginary-person yeah the op name in the list is only used as long as the method_tests for that op exist, so yes, please remove the op names from the `complex_list` as you port the tests.

File: torch/testing/_internal/common_methods_invocations.py [hunk 4]
```diff
@@ -602,7 +602,7 @@ def sample_inputs_addmm(op_info, device, dtype, requires_grad, **kwargs):
     else:
         return (input, )
 
-def sample_inputs_addbmm(op_info, device, dtype, requires_grad):
+def sample_inputs_addbmm(op_info, device, dtype, requires_grad, **kwargs):
```
Comment (imaginary-person) [inline torch/testing/_internal/common_methods_invocations.py:605]: I think the sample inputs would've to be changed, as per the changes brought by #53014.
eg. https://github.com/pytorch/pytorch/pull/55201#discussion_r608363961

These tests aren't running properly due to https://github.com/pytorch/pytorch/pull/54945#discussion_r607039433.

**Comment by imaginary-person** [comment]
Had a PR reverted because of a similar issue 😞

**Pull request closed without merging**